node-daemon/RPC_INTERFACE.md

```diff
@@ -0,0 +1,1170 @@
+
+Generate a block with the given transactions
+
+If `transactions` is `None`, the block will be generated with
+available transactions in the mempool
+
+
+Parameters:
+```
+{
+    "input_data": hex string,
+    "transactions": [ hex string, .. ],
+    "transaction_ids": [ hex string, .. ],
+    "packing_strategy": "FillSpaceFromMempool" OR "LeaveEmptySpace",
+}
+```
+
+Returns:
+```
+hex string
+```
+
+
+### Method `blockprod_e2e_public_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+hex string
+```
+
+
+### Method `blockprod_generate_block_e2e`
+
+Parameters:
+```
+{
+    "encrypted_input_data": [ number, .. ],
+    "public_key": hex string,
+    "transactions": [ hex string, .. ],
+    "transaction_ids": [ hex string, .. ],
+    "packing_strategy": "FillSpaceFromMempool" OR "LeaveEmptySpace",
+}
+```
+
+Returns:
+```
+hex string
+```
+
+
+## Module `test_functions`
+
+### Method `test_functions_genesis_pool_id`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string OR null
+```
+
+
+### Method `test_functions_genesis_private_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string OR null
+```
+
+
+### Method `test_functions_genesis_public_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string OR null
+```
+
+
+### Method `test_functions_genesis_vrf_private_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string OR null
+```
+
+
+### Method `test_functions_genesis_vrf_public_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string OR null
+```
+
+
+### Method `test_functions_new_private_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_public_key_from_private_key`
+
+Parameters:
+```
+{
+    "private_key_hex": string,
+}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_sign_message_with_private_key`
+
+Parameters:
+```
+{
+    "private_key_hex": string,
+    "message_hex": string,
+}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_verify_message_with_public_key`
+
+Parameters:
+```
+{
+    "public_key_hex": string,
+    "message_hex": string,
+    "signature_hex": string,
+}
+```
+
+Returns:
+```
+bool
+```
+
+
+### Method `test_functions_new_vrf_private_key`
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_vrf_public_key_from_private_key`
+
+Parameters:
+```
+{
+    "private_key_hex": string,
+}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_sign_message_with_vrf_private_key`
+
+Parameters:
+```
+{
+    "private_key_hex": string,
+    "epoch_index": number,
+    "random_seed": string,
+    "block_timestamp": number,
+}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_verify_then_get_vrf_output`
+
+Parameters:
+```
+{
+    "epoch_index": number,
+    "random_seed": string,
+    "vrf_data": string,
+    "vrf_public_key": string,
+    "block_timestamp": number,
+}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `test_functions_generate_transactions`
```

Comment:
Another example of things that have to be hidden

Reply:
The test functions are in a separate RPC module. These can be hidden easily by excluding the module from the output. The tool that generates the docs already has the `--dev` switch that makes it include the test functions module. I can easily re-run it without the switch and this module will be gone.